drivers/flash/nrf_qspi_nor.c

```diff
@@ -709,14 +773,14 @@ static int qspi_nrfx_configure(const struct device *dev)
 	}
 #endif
 
-#ifdef CONFIG_PM_DEVICE
+#if defined(CONFIG_PM_DEVICE) || defined(CONFIG_NORDIC_QSPI_NOR_IDLE_IN_DPD)
```

Comment:
I'm not happy either to see shortcuts around CONFIG_PM_DEVICE. What is the reason `CONFIG_PM_DEVICE` can't be used?

Reply:
this IDLE_IN_DPD support does not rely on CONFIG_PM_DEVICE (and in our application, we don't have CONFIG_PM_DEVICE enabled), so we need to define the DPD handlers in either case. 

I could build a macro for "USES_DPD" or similar, if that would make this better.

Reply:
why can't application enable `CONFIG_PM_DEVICE`?